### DISCUSSION TITLE
Switch to Default Host in Docker Environments

### DISCUSSION THREAD

**Discussion by kiwavi** [Kludex/uvicorn#2325] [Ideas] [answered]
Hello, 

Using uvicorn to run my fastAPI application and when I was dockerizing it I noticed that, without adding the "--host 0.0.0.0" for docker while running the server, I could not access localhost on host machine browser. This is because uvicorn defaults to 127.0.0.1. Is it possible to add a feature where uvicorn detects docker environments and switches to host 0.0.0.0? 

Thanks in advance.

**Comment by Kludex** [comment, marked as answer]
There's no way to do the detection properly.
